src/ensembl/src/content/app/species-selector/state/speciesSelectorActions.ts

```diff
@@ -158,42 +206,129 @@ export const handleSelectedSpecies: ActionCreator<
   dispatch(fetchAssemblies(genome_id));
 };
 
-export const commitSelectedSpecies = createStandardAction(
-  'species_selector/commit_selected_species'
-)();
+export const commitSelectedSpecies = createAction(
+  'species_selector/commit_selected_species',
+  (resolve) => {
+    return (itemName: string, committedSpecies: CommittedItem[]) => {
+      return resolve(
+        committedSpecies,
+        buildAnalyticsObject({
+          category: categories.ADD_SPECIES,
+          label: itemName,
+          action: 'select'
+        })
+      );
+    };
+  }
+);
 
 export const commitSelectedSpeciesAndSave: ActionCreator<
   ThunkAction<void, any, null, Action<string>>
 > = () => (dispatch, getState) => {
-  dispatch(commitSelectedSpecies());
   const committedSpecies = getCommittedSpecies(getState());
+  const selectedItem = getSelectedItem(getState());
+
+  if (!selectedItem) {
+    return;
+  }
+
+  const newCommittedSpecies = [
+    ...committedSpecies,
+    buildCommittedItem(selectedItem)
+  ];
+
+  const selectedItemName =
+    selectedItem.common_name || selectedItem.scientific_name;
+
+  dispatch(commitSelectedSpecies(selectedItemName, newCommittedSpecies));
+
   speciesSelectorStorageService.saveSelectedSpecies(committedSpecies);
 };
 
-export const toggleSpeciesUse = createStandardAction(
-  'species_selector/toggle_species_use'
-)<string>();
+export const toggleSpeciesUse = createAction(
+  'species_selector/toggle_species_use',
+  (resolve) => {
+    return (
+      speciesName: string,
+      currentStatus: string,
+      committedSpecies: CommittedItem[]
+    ) => {
+      return resolve(
+        committedSpecies,
+        buildAnalyticsObject({
+          category: categories.SELECTED_SPECIES,
+          label: speciesName,
+          action: currentStatus
+        })
+      );
+    };
+  }
+);
 
 export const toggleSpeciesUseAndSave: ActionCreator<
   ThunkAction<void, any, null, Action<string>>
 > = (genomeId: string) => (dispatch, getState) => {
-  dispatch(toggleSpeciesUse(genomeId));
   const committedSpecies = getCommittedSpecies(getState());
+  let speciesName = '';
+  let currentStatus = '';
+  committedSpecies.map((item) => {
+    if (item.genome_id === genomeId) {
+      speciesName = item.common_name || item.scientific_name;
+      currentStatus = item.isEnabled ? 'do_not_use' : 'use';
+      return {
+        ...item,
+        isEnabled: !item.isEnabled
+      };
+    }
+  });
+  dispatch(toggleSpeciesUse(speciesName, currentStatus, committedSpecies));
   speciesSelectorStorageService.saveSelectedSpecies(committedSpecies);
 };
 
-export const deleteSpecies = createStandardAction(
-  'species_selector/delete_species'
-)<string>();
+export const deleteSpecies = createAction(
+  'species_selector/delete_species',
+  (resolve) => {
+    return (speciesName: string, committedSpecies: CommittedItem[]) => {
```

Comment:
As above, speciesName is easily calculable from committedSpecies.

Reply:
committedSpecies will not have the species anymore as we delete it beforehand.